src/environments.js

```diff
@@ -216,13 +216,20 @@ defineEnvironment("aligned", {
     let res = {
         type: "array",
         cols: [],
+        rowsAreLines: true,
     };
-    res = parseArray(context.parser, res);
+    res = parseArray(context.parser, res, "display");
+    // Count number of columns = maximum number of cells in each row.
+    // At the same time, prepend empty group {} at beginning of every second
+    // cell in each row (starting with second cell) so that operators become
+    // binary.
```

Comment:
I just added documentation here, because it took me a long time to figure out why the existing code was adding empty groups.  I checked and this is indeed how it's implemented in amsmath, [specifically in `\start@aligned`](http://tug.ctan.org/macros/latex/required/amslatex/math/amsmath.pdf#page=46).